app/controllers/posts_controller.rb

```diff
@@ -6,7 +6,41 @@ def index
 
   def show
     @post = Post.find(params[:id])
-    @user = @post.author
+    @user = current_user
     @comments = @post.comments
   end
+
+  def new
+    @post = Post.new
+    @user = current_user
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+    if @post.save
+      puts @post
+      redirect_to user_posts_path(current_user)
+    else
+      render :new
+    end
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    Like.new(author: current_user, post: @post).save
+    redirect_to user_post_path(current_user, @post)
+  end
+
+  def comment
+    @post = Post.find(params[:id])
+    p params
+    @comment = Comment.new(author: current_user, post: @post, text: params[:text])
+    @comment.save
+    redirect_to user_post_path(current_user, @post)
+  end
```

Comment:
- [ ] Let's refactor this section a bit. Please move the like and comment logic to separate controllers(they will have their own CRUD logic in a later milestone) with the required actions. Also, please don't forget to change the routes config accordingly.

Reply:
Done :100: